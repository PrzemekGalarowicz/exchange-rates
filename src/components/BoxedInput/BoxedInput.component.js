import React from 'react';
import PropTypes from 'prop-types';
import { Form } from 'react-bulma-components';

const { Input, Field, Select, Control, Label } = Form;

const BoxedInput = ({
  size,
  selectLabel,
  selectDisabled,
  selectLoading,
  selectDefaultValue,
  selectOptions,
  inputLabel,
  inputMin,
  inputMax,
  inputDisabled,
  inputType,
  onChange
}) => (
  <Field kind="addons" alignItems="flex-end">
    <Control>
      <Field>
        {selectLabel && <Label>{selectLabel}</Label>}
        <Control>
          <Select
            size={size}
            loading={selectLoading}
            disabled={selectDisabled}
            defaultValue={selectDefaultValue}>
            {selectOptions?.map((item, i) => (
              <option key={i}>{item}</option>
            ))}
          </Select>
        </Control>
      </Field>
    </Control>
    <Control fullwidth>
      <Field alignItems="flex-end">
        {inputLabel && <Label>{inputLabel}</Label>}
        <Control>
          <Input
            size={size}
            type={inputType}
            min={inputMin}
            max={inputMax}
            disabled={inputDisabled}
            onChange={onChange}
          />
          {/* TODO: Add some loading state for inputs: {inputLoading && ???} */}
        </Control>
      </Field>
    </Control>
  </Field>
);

BoxedInput.defaultProps = {
  size: 'medium',
  selectDisabled: false,
  selectLoading: false,
  selectDefaultValue: '',
  inputDisabled: false,
  inputLoading: false,
  inputType: 'text'
};

BoxedInput.propTypes = {
  size: PropTypes.string,
  selectLabel: PropTypes.string,
  selectDisabled: PropTypes.bool,
  selectLoading: PropTypes.bool,
  selectDefaultValue: PropTypes.string,
  selectOptions: PropTypes.array.isRequired,
  inputLabel: PropTypes.string,
  inputType: PropTypes.string,
  inputMin: PropTypes.string,
  inputMax: PropTypes.string,
  inputLoading: PropTypes.bool,
  inputDisabled: PropTypes.bool,
  onChange: PropTypes.func
};

export default BoxedInput;
