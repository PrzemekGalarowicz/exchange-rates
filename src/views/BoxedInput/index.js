export { default } from './BoxedInput.component.js';
