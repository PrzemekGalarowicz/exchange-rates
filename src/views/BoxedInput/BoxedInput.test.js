import React from 'react';
import renderer from 'react-test-renderer';
import BoxedInput from './BoxedInput.component';

const defaultProps = {
  size: 'large',
  selectLabel: 'Select label',
  selectDisabled: false,
  selectLoading: false,
  selectDefaultValue: 'PLN',
  selectOptions: ['PLN', 'USD', 'EUR'],
  onSelectChange: jest.fn(),
  inputLabel: 'Input label',
  inputMin: 0,
  inputMax: 100,
  inputDisabled: false,
  inputType: 'number',
  onInputChange: jest.fn()
};

describe('<BoxedInput />', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<BoxedInput {...defaultProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without size', () => {
    const updatedProps = {
      ...defaultProps,
      size: undefined
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without selectLabel', () => {
    const updatedProps = {
      ...defaultProps,
      selectLabel: undefined
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without selectDefaultValue', () => {
    const updatedProps = {
      ...defaultProps,
      selectDefaultValue: undefined
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with selectDisabled state', () => {
    const updatedProps = {
      ...defaultProps,
      selectDisabled: true
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with selectLoading state', () => {
    const updatedProps = {
      ...defaultProps,
      selectLoading: true
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without inputLabel', () => {
    const updatedProps = {
      ...defaultProps,
      inputLabel: undefined
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without inputMin', () => {
    const updatedProps = {
      ...defaultProps,
      inputMin: undefined
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without inputMax', () => {
    const updatedProps = {
      ...defaultProps,
      inputMax: undefined
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without inputType', () => {
    const updatedProps = {
      ...defaultProps,
      inputType: undefined
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with inputDisabled state', () => {
    const updatedProps = {
      ...defaultProps,
      inputDisabled: true
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with inputLoading state', () => {
    const updatedProps = {
      ...defaultProps,
      inputLoading: true
    };
    const tree = renderer.create(<BoxedInput {...updatedProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
