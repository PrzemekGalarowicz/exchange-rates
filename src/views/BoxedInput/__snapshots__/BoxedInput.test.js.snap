// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BoxedInput /> renders correctly 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly with inputDisabled state 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={true}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly with inputLoading state 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly with selectDisabled state 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={true}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly with selectLoading state 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large is-loading"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly without inputLabel 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly without inputMax 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly without inputMin 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly without inputType 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly without selectDefaultValue 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue=""
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly without selectLabel 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <div
        className="control"
      >
        <div
          className="select is-large"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-large"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`<BoxedInput /> renders correctly without size 1`] = `
<div
  className="field has-addons is-align-items-flex-end"
>
  <div
    className="control"
  >
    <div
      className="field"
    >
      <label
        className="label"
      >
        Select label
      </label>
      <div
        className="control"
      >
        <div
          className="select is-medium"
        >
          <select
            defaultValue="PLN"
            disabled={false}
            onChange={[MockFunction]}
          >
            <option>
              PLN
            </option>
            <option>
              USD
            </option>
            <option>
              EUR
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
  <div
    className="control is-expanded"
  >
    <div
      className="field is-align-items-flex-end"
    >
      <label
        className="label"
      >
        Input label
      </label>
      <div
        className="control"
      >
        <input
          className="input is-medium"
          disabled={false}
          max={100}
          min={0}
          onChange={[MockFunction]}
          type="number"
        />
      </div>
    </div>
  </div>
</div>
`;
